File: Python/lab11/OrderService.py
from Order import Order

class OrderService:
    def __init__(self, order: Order):
        self.order = order
        self.free = False

    def make_order(self):
        with open('orders.txt', 'r') as file:
            text = file.read()

        self.free = len(text) > 5

        if self.free:
            self.order.create_order()
            print("Your order has been created.")
        else:
            print("Oops, something went wrong.")

File: Java/lab11/src/main/java/org/example/OrderService.java
package org.example;

import java.io.FileInputStream;
import java.io.IOException;
import java.util.Scanner;

public class OrderService {
    Order order;
    boolean free;
    public OrderService(Order order) {
        this.order = order;
    }
    public void makeOrder(){
        StringBuilder text= new StringBuilder();
        try(FileInputStream read = new FileInputStream("orders.txt")) {
            Scanner scan = new Scanner(read);
            while (scan.hasNextLine()) {
                text.append(scan.nextLine());
            }
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
        free= text.length() > 5;
        if(free){
            order.createOrder();
            System.out.println("your order create");
        }else {
            System.out.println("Ops, something went wrong");
        }
    }
}
